### ISSUE TITLE
Trident 20.01 on OpenShift modifies the 'privileged' SCC

### ISSUE THREAD

**Issue by e-minguez** [NetApp/trident#396]
**Describe the bug**
The best practice dictates that the default SCCs in OpenShift shouldn't be modified but create new ones if needed (https://docs.openshift.com/container-platform/4.4/authentication/managing-security-context-constraints.html)

I've deployed trident 20.01 on my OpenShift cluster and as you can see, the privileged SCC is modified as the trident-csi service account is added.

I've extracted the scc into a file pre and post installing trident and those are the results:

```
$ yq compare --prettyPrint 4.3-ootb/clusterwide/securitycontextconstraints.security.openshift.io.yaml 4.3-trident/clusterwide/securitycontextconstraints.security.openshift.io.yaml 'items.(metadata.name==privileged)'
 allowHostDirVolumePlugin: true
 allowHostIPC: true
 allowHostNetwork: true
 allowHostPID: true
 allowHostPorts: true
 allowPrivilegeEscalation: true
 allowPrivilegedContainer: true
 allowedCapabilities:
 - '*'
 allowedUnsafeSysctls:
 - '*'
 apiVersion: security.openshift.io/v1
 defaultAddCapabilities: null
 fsGroup:
   type: RunAsAny
 groups:
 - system:cluster-admins
 - system:nodes
 - system:masters
 kind: SecurityContextConstraints
 metadata:
   annotations:
     kubernetes.io/description: 'privileged allows access to all privileged and host
       features and the ability to run as any user, any group, any fsGroup, and with
       any SELinux context.  WARNING: this is the most relaxed SCC and should be used
       only for cluster administration. Grant with caution.'
   creationTimestamp: "2020-05-04T10:28:52Z"
-  generation: 1
+  generation: 2
   name: privileged
-  resourceVersion: "6489"
+  resourceVersion: "88968"
   selfLink: /apis/security.openshift.io/v1/securitycontextconstraints/privileged
   uid: b000901d-f681-4df8-9d3b-876b1cb4e90d
 priority: null
 readOnlyRootFilesystem: false
 requiredDropCapabilities: null
 runAsUser:
   type: RunAsAny
 seLinuxContext:
   type: RunAsAny
 seccompProfiles:
 - '*'
 supplementalGroups:
   type: RunAsAny
 users:
 - system:admin
 - system:serviceaccount:openshift-infra:build-controller
+- system:serviceaccount:trident:trident-csi
 volumes:
 - '*'
```

**Environment**
- OpenShift 4.3.8
- Trident version: 20.01
- Trident installation flags used: `--generate-custom-yaml -n trident` then `install -d -n trident --use-custom-yaml
` (with no modifications)
- Container runtime: `cri-o://1.16.3-28.dev.rhaos4.3.git9aad8e4.el8`
- Kubernetes version: `v1.16.2`
- Kubernetes orchestrator: `4.3.8`
- Kubernetes enabled feature gates:
```
$ oc get featuregates/cluster -o yaml
apiVersion: config.openshift.io/v1
kind: FeatureGate
metadata:
  annotations:
    release.openshift.io/create-only: "true"
  creationTimestamp: "2020-05-04T10:21:46Z"
  generation: 1
  name: cluster
  resourceVersion: "1615"
  selfLink: /apis/config.openshift.io/v1/featuregates/cluster
  uid: 5e2e5a66-e5ba-4fce-b6ae-bfce8688a43e
spec: {}
```
- OS: `Red Hat Enterprise Linux CoreOS 43.81.202003191953.0`
- NetApp backend types: `NFS`
- Other:

**To Reproduce**
* Get the sccs before deploying trident:
```
oc get scc -o yaml >> pre.yaml
```
* Install trident
```
./tridentctl install ...
```

* Get the sccs after deploying trident:
```
oc get scc -o yaml >> post.yaml
```

* Observe the differences

**Expected behavior**
The default SCCs are not modified but a new one is added if needed.

**Additional context**
See here for more context https://access.redhat.com/solutions/4972291

**Comment by clintonk** [comment]
Hello, @e-minguez.  This was reported in #374 and fixed in Trident 20.04.

**Comment by e-minguez** [comment]
> Hello, @e-minguez. This was reported in #374 and fixed in Trident 20.04.

Great!, closing it now then. Thanks!
